app/src/main/AndroidManifest.xml

```diff
@@ -54,6 +54,9 @@
         <category android:name="android.intent.category.LAUNCHER" />
       </intent-filter>
     </activity>
+    <activity
+      android:name=".activities.OnboardingActivity"
+      android:exported="false" />

```

Comment:
I don't think it is necessary to create another activity just for this purpose. Currently, the `SplashActivity` simply navigates to the `MainActivity` i.e. it does nothing else.

I suggest that you remove the splash activity and set the `CATEGORY_LAUNCHER` flag to the onboarding activity (basically, the onboarding activity should be the first activity to open, instead of the splash activity). Then, if the tools aren't installed or if it the onboarding activity was not shown before, show the onboarding app intro and proceed. Otherwise, navigate to the main activity and finish the onboarding activity.

Simply, when the IDE is launched, show onboarding activity, then :

```
if (are_tools_installed && has_shown_intro) {
    navigate_to_main_activity();
    return;
}

show_app_intro();

```